src/api.rs

```diff
@@ -1369,6 +1495,67 @@ impl Client {
     }
 }
 
+fn start_sso_callback_server(
+    listener: &TcpListener,
+    state: &str,
+) -> Result<String> {
```

Comment:
in general in this function, i'm not particularly comfortable with hand-parsing and writing raw http here. we already pull in enough libraries that we can do this in a less fragile way (i do kind of agree that we don't necessarily need a full blown web server here, but at least using `hyper` or `http` or something like that to parse and generate the http would be a lot less fragile)